wasm.js

```
@@ -0,0 +1,185 @@
+/*
+ * This file is part of the 0chain @zerochain/0chain distribution (https://github.com/0chain/client-sdk).
+ * Copyright (c) 2018 0chain LLC.
+ *
+ * 0chain @zerochain/0chain program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+const utils = require('./utils');
+
+// Initially, the __zcn_wasm__ object will be
+const g = global || window || self;
+
+/**
+ * bridge is an easier way to refer to the Go WASM object.
+ */
+const bridge = g.__zcn_wasm__;
+
+async function blsSign (hash) {
+  if (!bridge.bls?.secretKey) {
+    const msg = 'err: bls.secretKey is not initialized';
+    console.error(msg);
+    return msg;
+  }
+
+  try {
```

In line with the above comment, we should not use try/catch, but let the lib throw the errors it finds. The client must be responsible for catching and handling.